lib/Dialect/TritonGPU/Transforms/LoopScheduling.cpp

```diff
@@ -35,305 +18,108 @@ namespace mlir {
-      unsigned vec = axisInfoAnalysis.getPtrContiguity(ptr);
-      if (auto mask = loadOp.getMask())
-        vec = std::min<unsigned>(vec, axisInfoAnalysis.getMaskAlignment(mask));
-
-      auto tensorTy = dyn_cast<RankedTensorType>(ptr.getType());
-      if (!tensorTy)
-        continue;
-      auto ty =
-          cast<tt::PointerType>(tensorTy.getElementType()).getPointeeType();
-      unsigned width = vec * ty.getIntOrFloatBitWidth();
-
-      // We do not pipeline all loads for the following reasons:
-      // 1. On nvidia GPUs, cp.async's cp-size can only be 4, 8, or 16.
-      // 2. It's likely that pipling small loads won't offer much performance
-      //    improvement and may even hurt performance by increasing register
-      //    pressure.
-      LDBG("Load " << *loadOp << " has width " << width);
-      if (width < 32)
-        continue;
-    }
-
-    bool hasSharedEncoding = false;
-    if (use->hasTrait<OpTrait::DotLike>()) {
-      auto mmaLoadType = getMMALoadType(op);
-      auto dot = dyn_cast<tt::DotOp>(use);
-      auto warpGroupDot = dyn_cast<ttng::WarpGroupDotOp>(use);
-      bool isMMAv3Shared = mmaLoadType == MMALoadType::SharedV3;
-      bool isMMAv3Registers =
-          (mmaLoadType == MMALoadType::Registers) && warpGroupDot;
-
-      if (isMMAv3Shared) {
-        hasSharedEncoding = true;
-      } else if (isa<tt::ExperimentalDescriptorLoadOp>(op)) {
-        hasSharedEncoding = true;
-      } else if (isMMAv3Registers || dot) {
-        // FIXME: if we have a better solution in handling incompatible shared
-        // encoding, we can simplify the logic here by checking if all users are
-        // dot encoding. Fow now, getSharedEncIfAllUsersAreDotEnc will be used
-        // during both scheduling and lowering.
-        bool incompatible = false;
-        auto sharedEncoding =
-            getSharedEncIfAllUsersAreDotEnc(op->getResult(0), incompatible)
-                .value_or(nullptr);
-        hasSharedEncoding = sharedEncoding != nullptr;
-        // If we can't agree on a shared encoding skip pipelinig the load.
-        if (incompatible)
-          continue;
-      }
-    } else if (auto loadOp = dyn_cast<tt::LoadOp>(use)) {
-      // The use of this loadOp is another loadOp. If the use is not in the
-      // loadsToPipeline already, it means that the use is not valid for
-      // pipelining for some reason. We should skip this loadOp, too. Note that
-      // we have an assumption that distAndUse.second (i.e. the use of this
-      // loadOp) has already be processed in a previous loop iteration. This
-      // assumption is held by how loadOpsToIndirectionLevelAndUse recursively
-      // collects loadOpToIndLevelAndUse using DFS.
-      if (loadsToPipeline.count(loadOp) == 0) {
+  // If no latency ops, nothing to schedule
+  if (latOps.empty())
+    return CoarseSchedule(0);
+
+  // Compute the longest path to the yield for each operation reachable
+  // from any latency operation.
```

Comment:
When defining the longest path, do ops have latency 0 if they are not in opLatency?

Reply:
Yes, that's the idea